activesupport/lib/active_support/core_ext/time/compatibility.rb

```diff
@@ -1,5 +1,12 @@
 require "active_support/core_ext/date_and_time/compatibility"
+require "active_support/core_ext/module/remove_method"
 
 class Time
-  prepend DateAndTime::Compatibility
+  include DateAndTime::Compatibility
+
+  remove_possible_method :to_time
```

Comment:
I'm fuzzy on the benefits of overshadowing vs `undef_method`. So this may not be necessary.